### PR TITLE
Very basic pagination

### PR DESCRIPTION
**Pull request by dphiffer**
Re: #29 adds pagination to the shlink manager interface

Screenshot 2024-05-31 at 5 15 42 PM

### CODE REVIEW

File: src/js/manager.js [hunk 1]
```diff
@@ -42,13 +55,16 @@ class SmolLinksManager {
 
 			let form = document.querySelector('.smol-links-create');
 			form.addEventListener('submit', this.createShlink.bind(this));
+
+			let pagination = document.querySelector('.smol-links-pagination');
+			pagination.addEventListener('change', this.updatePage.bind(this))
 		} catch (err) {
```
Comment (BatMiles) [inline src/js/manager.js:64]: this catch doesn't seem to be doing anything now, can it disappear?

File: src/js/manager.js [hunk 2]
```diff
@@ -22,18 +23,30 @@ class SmolLinksManager {
...
-					(response.error ||
-						'something went wrong loading shlinks. Try again?');
+				html = `<div class="smol-links-error">
+					<strong>Error: </strong>
+					${(response.error ||
+						'something went wrong loading shlinks. Try again?')}
+				</div>`;
 			} else {
 				html = this.getListHTML(response.shlink.shortUrls.data);
+				html += this.getPaginationHTML(response.shlink.shortUrls.pagination);
```
Comment (BatMiles) [inline src/js/manager.js]:
```suggestion
				html = this.getPaginationHTML(response.shlink.shortUrls.pagination) + html;
```

to put pagination at the top of the list where people are more likely to see it

File: src/css/manager.scss [hunk 3]
```diff
@@ -226,3 +226,7 @@
 		margin-bottom: 0;
 	}
 }
+
+.smol-links-error {
+	padding: 20px 0;
+}
```
Comment (BatMiles) [inline src/css/manager.scss:236]: if pagination moves to the top, a little bit of margin:
```
.smol-links-pagination {
	margin-top: 5px;
}
```

**Review by BatMiles — APPROVED** [review]
Woo!

File: src/js/manager.js (hunk 1 above)
Comment (dphiffer) [reply src/js/manager.js:64]: ah thanks, fixed it!

**Pull request merged**